common/paramsbuilder/params.go

```diff
@@ -78,20 +79,54 @@ func (p *Workspace) WithWorkspace(workspaceRef string) {
 	p.Name = workspaceRef
 }
 
+func (p *Workspace) Substitution() map[string]string {
+	return map[string]string{"workspace": p.Name}
+}
+
 // Module params adds suffix to URL controlling API versions.
 // This is relevant where there are several APIs for different product areas or sub-products, and the APIs
 // are versioned differently or have different ways of constructing URLs from object names.
 type Module struct {
-	Suffix string
+	Suffix    string
+	supported []APIModule
+	fallback  *APIModule
```

Comment:
Does a fallback module have to be in the supported list?

Reply:
Yes I validate the fallback too to be in the supported list. 
https://github.com/amp-labs/connectors/blob/730046a36c2e81e96b83b8b9292e516e84151bfe/common/paramsbuilder/params.go#L98-L110

For hubspot it was hard coded deep inside but now this intention will be explicit. Note: supported/fallback are internal fields that should be set by us.
https://github.com/amp-labs/connectors/blob/730046a36c2e81e96b83b8b9292e516e84151bfe/hubspot/params.go#L59-L64